shell/platform/tizen/channels/text_input_channel.cc

```diff
@@ -57,6 +57,44 @@ static bool IsASCIIPrintableKey(char c) {
   return false;
 }
 
+static bool TextInputTypeToEcoreIMFInputPanelLayout(
+    std::string text_input_type,
+    Ecore_IMF_Input_Panel_Layout& panel_layout) {
+  bool ret = false;
+  if (text_input_type == "TextInputType.text" ||
+      text_input_type == "TextInputType.multiline") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
+    ret = true;
+  } else if (text_input_type == "TextInputType.number") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NUMBER;
+    ret = true;
+  } else if (text_input_type == "TextInputType.phone") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_PHONENUMBER;
+    ret = true;
+  } else if (text_input_type == "TextInputType.datetime") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_DATETIME;
+    ret = true;
+  } else if (text_input_type == "TextInputType.emailAddress") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_EMAIL;
+    ret = true;
+  } else if (text_input_type == "TextInputType.url") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_URL;
+    ret = true;
+  } else if (text_input_type == "TextInputType.visiblePassword") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_PASSWORD;
+    ret = true;
+  } else if (text_input_type == "TextInputType.name" ||
+             text_input_type == "TextInputType.address") {
+    FT_LOGW(
+        "Actual requested text input type is [%s], but select "
+        "TextInputType.text as fallback type",
+        text_input_type.c_str());
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
+    ret = true;
+  }
+  return ret;
```

Comment:
How about adding an else statement:

```suggestion
  if (text_input_type == "TextInputType.text" ||
      text_input_type == "TextInputType.multiline") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
  } else if (text_input_type == "TextInputType.number") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NUMBER;
  } else if (text_input_type == "TextInputType.phone") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_PHONENUMBER;
  } else if (text_input_type == "TextInputType.datetime") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_DATETIME;
  } else if (text_input_type == "TextInputType.emailAddress") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_EMAIL;
  } else if (text_input_type == "TextInputType.url") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_URL;
  } else if (text_input_type == "TextInputType.visiblePassword") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_PASSWORD;
  } else if (text_input_type == "TextInputType.name" ||
             text_input_type == "TextInputType.address") {
    FT_LOGW(
        "Actual requested text input type is [%s], but select "
        "TextInputType.text as fallback type",
        text_input_type.c_str());
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
  } else {
    return false;
  }
  return true;
```

or just returning `ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL` if the requested type is unsupported:

```suggestion
  if (text_input_type == "TextInputType.text" ||
      text_input_type == "TextInputType.multiline") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
  } else if (text_input_type == "TextInputType.number") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_NUMBER;
  } else if (text_input_type == "TextInputType.phone") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_PHONENUMBER;
  } else if (text_input_type == "TextInputType.datetime") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_DATETIME;
  } else if (text_input_type == "TextInputType.emailAddress") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_EMAIL;
  } else if (text_input_type == "TextInputType.url") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_URL;
  } else if (text_input_type == "TextInputType.visiblePassword") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_PASSWORD;
  } else {
    FT_LOGW(
        "Actual requested text input type is [%s], but select "
        "TextInputType.text as fallback type",
        text_input_type.c_str());
    return ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
  }
```

Reply:
Thanks! I choose first one :smile: